hack/list-images.sh

```diff
@@ -0,0 +1,135 @@
+            kommanderAppManagementKubetoolsImageRepository='' \
+            kommanderAppManagementWebhookImageRepository='' \
+            kommanderAppManagementConfigAPIImageRepository='' \
+            tfaName=unused \
+            notPopulatedAnywhereAsThisIsOnlyForAirgappedBundle=unused \
+            caIssuerName=unused \
+            kommanderChartVersion="${kommanderChartVersion:-}"
+
+IMAGES_FILE="$(realpath "$(mktemp .helm-list-images-XXXXXX)")"
+readonly IMAGES_FILE
+trap_add "rm --force ${IMAGES_FILE}" EXIT
+
+for dir in $(find . -type f -name "*.yaml" -print0 | xargs --null --max-lines=1 --no-run-if-empty -- grep --files-with-matches '^kind: HelmRelease' | grep --only-matching "\(.*\)/" | sort --unique); do
+  pushd "${dir}" &>/dev/null
+
+  while IFS= read -r hr; do
+    pushd "$(dirname "${hr}")" &>/dev/null
+    extra_args=()
+    defaults_cm=""
+    if [ -f 'defaults/cm.yaml' ]; then
+      defaults_cm='defaults/cm.yaml'
+    elif [ -f '../defaults/cm.yaml' ]; then
+      defaults_cm='../defaults/cm.yaml'
+    fi
+    if [ -n "${defaults_cm}" ]; then
+      temp_values="$(mktemp .helm-list-images-XXXXXX)"
+      trap_add "rm --force $(realpath "${temp_values}")" EXIT
+      envsubst -no-unset -no-digit -i "${defaults_cm}" | gojq --yaml-input --raw-output '.data.["values.yaml"]' | sed '/---/d' >"${temp_values}"
+      extra_args+=('--values' "${temp_values}")
+    fi
+
+    if [ -f 'list-images-values.yaml' ]; then
+      extra_args+=('--values' 'list-images-values.yaml')
+    fi
+    if [ -f 'extra-images.txt' ]; then
+      extra_args+=('--extra-images-file' 'extra-images.txt')
+    fi
+
+    envsubst -no-unset -no-digit -i "$(basename "${hr}")" | \
+      gojq --yaml-input --raw-output --arg repoRoot "${REPO_ROOT}" \
+        $'select(.spec.chart.spec.sourceRef.name != null) |
+          if .spec.chart.spec.sourceRef.kind == "HelmRepository" then
+            (.spec.chart.spec.sourceRef.name | gsub("\\\."; "-"))+"/"+.spec.chart.spec.chart+" --chart-version="+.spec.chart.spec.version
+          elif .spec.chart.spec.sourceRef.kind == "GitRepository" then
+            $repoRoot+"/"+.spec.chart.spec.chart
+          end' | \
+      xargs --max-lines=1 --no-run-if-empty -- helm list-images --unique "${extra_args[@]}" >>"${IMAGES_FILE}"
+    popd &>/dev/null
+  done < <(grep --recursive --max-count=1 --files-with-matches '^kind: HelmRelease')
+  popd &>/dev/null
+done
+
+gojq --yaml-input --raw-output 'select(.kind | test("^(?:Deployment|Job|CronJob|StatefulSet|DaemonSet)$")) |
+                                .spec.template.spec |
+                                (select(.containers != null) | .containers[].image), (select(.initContainers != null) | .initContainers[].image)' \
+                                ./services/kommander-flux/*/templates/* \
+                                ./services/kube-prometheus-stack/*/etcd-metrics-proxy/* \
+                                ./services/velero/*/{pre,post}-install/* \
```

Comment:
Can you add a comment why these are hardcoded and when a list should be changed? Will this fail if the file structure changes?

Reply:
Added a couple of comment.